packages/context/src/inject.ts

```diff
@@ -413,6 +409,7 @@ function resolveAsGetter(
     return ctx.get(bindingSelector, {
       session: forkedSession,
       optional: injection.metadata.optional,
+      asProxyWithInterceptors: injection.metadata.asProxyWithInterceptors,
```

Comment:
Now that `InjectionMetadata` are extending `ResolutionOptions`, I'd prefer to avoid the need to copy subset of metadata properties and having to update this block whenever we add a new resolution option. I am proposing to  pass all properties from `injection.metadata` instead:

```ts
return ctx.get(bindingSelector, {
  ...(injection.metadata),
  session: forkedSession
});
```